reporter/iface.go

```diff
@@ -51,20 +51,27 @@ type TraceReporter interface {
 }
 
 type ExecutableOpener = func() (process.ReadAtCloser, error)
+type ExecutableMetadataArgs struct {
+	FileID            libpf.FileID
+	FileName          string
+	GnuBuildID        string
+	DebuglinkFileName string
+	Interp            libpf.InterpreterType
+	Open              ExecutableOpener
+}
 
 type SymbolReporter interface {
 	// ReportFallbackSymbol enqueues a fallback symbol for reporting, for a given frame.
 	ReportFallbackSymbol(frameID libpf.FrameID, symbol string)
 
-	// ExecutableMetadata accepts a fileID with the corresponding filename
+	// ExecutableMetadata accepts a FileID with the corresponding filename
```

Comment:
When at it, maybe you can reword the sentence!? Whether an implementation caches data or not is not relevant here.